source/lib/course-search/update-course-storage.js

```diff
@@ -59,5 +59,15 @@ async function loadCurrentTermsFromServer(): Promise<Array<TermType>> {
 async function storeTermCoursesFromServer(term: TermType) {
 	const url = COURSE_DATA_PAGE + term.path
 	const resp: Array<CourseType> = await fetchJson(url).catch(() => [])
-	storage.setTermCourseData(term.term, resp)
+	const formattedTermData = formatRawData(resp)
+	storage.setTermCourseData(term.term, formattedTermData)
+}
+
+function formatRawData(rawData: Array<CourseType>): Array<CourseType> {
+	return rawData.map(course => {
+		const open = course.enroll < course.max
+		let formattedCourse = course
+		course.open = open
+		return formattedCourse
```

Comment:
```js
let isFull = course.enroll < course.max
return {full: isFull, ...course}
```

Otherwise, you're just essentially grabbing a pointer to an object and mutating it.

Speaking of, we should make a second `type` – RawCourseType or something – to separate what we add from what the data provides.

Reply:
I believe you meant 
```javascript
let isFull = course.enroll >= course.max
```

Reply:
However...I think it makes more sense that when the user toggles this filter "on", that they will now only see classes with space available. Thus, having isFull as the key for the ToggleFilter would be counterintuitive. Could we just change `open` to `spaceAvailable`?

Reply:
That sounds fine to me. We should also think about exposing both "status: open" and "hasSpaceAvailable" in the filter, as they do represent different states.

Reply:
As it stands, they each have their own filter. "Enrollment" corresponds to `spaceAvailable` and "Status" corresponds to `status`. I can't think of a better name for Status, but I feel like there should be one.

Reply:
I think we should leave Status as Status, for consistency with SIS if nothing else. 

I didn't realize there was already a separate filter for status, my apologies.